src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -182,6 +185,10 @@ public class RSessionProvider : IRSessionProvider {
         }
 
         public async Task<bool> TrySwitchBrokerAsync(string name, string path = null, CancellationToken cancellationToken = default(CancellationToken)) {
+            if (_disposeToken.IsDisposed) {
+                return false;
+            }
+
             await TaskUtilities.SwitchToBackgroundThread();
 
             var brokerClient = CreateBrokerClient(name, path);
```

Comment:
`TrySwitchBrokerAsync` appears to be called from 2 places in the Connection Manager - technically we can end up with two concurrent brokerClient here